tests/DCPS/Thrasher/ParticipantTask.cpp

```diff
@@ -17,21 +17,61 @@
 #include <dds/DCPS/transport/framework/TransportConfig.h>
 #include <dds/DCPS/transport/framework/TransportInst.h>
 
+#include <dds/DCPS/RTPS/RtpsDiscovery.h>
+
 #include "ParticipantTask.h"
 #include "ProgressIndicator.h"
 #include "FooTypeTypeSupportImpl.h"
 
 ParticipantTask::ParticipantTask(const std::size_t& samples_per_thread)
   : samples_per_thread_(samples_per_thread)
-#ifdef OPENDDS_SAFETY_PROFILE
   , thread_index_(0)
-#endif
 {
 }
 
 ParticipantTask::~ParticipantTask()
 {}
 
+namespace {
+
+int
+wait_match(const DDS::DataWriter_var& writer,
+           unsigned int num_readers)
+{
+  DDS::StatusCondition_var condition = writer->get_statuscondition();
+  condition->set_enabled_statuses(DDS::PUBLICATION_MATCHED_STATUS);
+  DDS::WaitSet_var ws = new DDS::WaitSet;
+  ws->attach_condition(condition);
+  DDS::ConditionSeq conditions;
+  DDS::PublicationMatchedStatus ms = { 0, 0, 0, 0, 0 };
+  DDS::Duration_t timeout = { 3, 0 };
+  DDS::ReturnCode_t stat;
+  do {
```

Comment:
```suggestion
  while (true) {
```
Also in Subscriber.cpp -- should these be combined?

Reply:
Yes, and probably for more than these two tests.